python/pyarrow/dataset.py

```diff
@@ -682,3 +683,68 @@ def dataset(source, schema=None, format=None, filesystem=None,
             'Expected a path-like, list of path-likes or a list of Datasets '
             'instead of the given type: {}'.format(type(source).__name__)
         )
+
+
+def _ensure_write_partitioning(scheme):
+    if scheme is None:
+        scheme = partitioning(pa.schema([]))
+    if not isinstance(scheme, Partitioning):
+        # TODO support passing field names, and get types from schema
+        raise ValueError("partitioning needs to be actual Partitioning object")
+    return scheme
+
+
+def write_dataset(data, base_dir, format=None, partitioning=None, schema=None,
+                  filesystem=None, use_threads=True):
+    """
+    Write a dataset to a given format and partitioning.
+
+    Parameters
+    ----------
+    data : FileSystemDataset, Table/RecordBatch, or list of Table/RecordBatch
+        The data to write. This can be a FileSystemDataset instance or
```

Comment:
```suggestion
    data : Dataset, Table/RecordBatch, or list of Table/RecordBatch
        The data to write. This can be a Dataset instance or
```
(A UnionDataset could also be written, for example concatenating a Parquet and a CSV dataset into a single IPC dataset)